src/hotspot/share/memory/heap.cpp

```diff
@@ -553,7 +553,7 @@ int CodeHeap::allocated_segments() const {
 
 size_t CodeHeap::allocated_capacity() const {
   // size of used heap - size on freelist
-  return segments_to_size(_next_segment - _freelist_segments);
+  return segments_to_size(Atomic::load_acquire(&_next_segment) - _freelist_segments);
```

Comment:
```suggestion
  size_t next_segment = Atomic::load_acquire(&_next_segment);
  return segments_to_size(next_segment - _freelist_segments);
```
Reason: C++ can evaluate subexpressions in any order.